rules.py

```diff
@@ -829,24 +829,26 @@ def check(self):
        
 class DuplicateContestNames(base.TreeRule):
     """Check that the file contains unique ContestNames.
-	Add Warning if duplicate ContestName found.
-    """
-    names_lines={} # Mapping for names and sourcelines.
+	Add Warning if duplicate ContestName found."""
+
+    names_lines = {} # Mapping for names and sourcelines.
     error_log = []
     def check(self):
         for event, element in etree.iterwalk(
                 self.election_tree):
-            if(element.tag!='Contest'):
+	    tag = self.strip_schema_ns(element)
+            if tag != 'Contest':
                 continue
             else:
-                name=element.find("Name")
-                if name.text is not None and name.text!="":
-                    # Add names and sourcelines as key and list of values. Ideally 1 sourceline, if duplicate then list of sourcelines.
-                    self.names_lines.setdefault(name.text,[]).append(name.sourceline)
+                name = element.find("Name")
+                if name is not None and name.text:
+                    """Add names and sourcelines as key and list of values.
+		    Ideally 1 sourceline, if duplicate then list of sourcelines."""
```

Comment:
I'd suggest tweaking this so you're creating a dict which maps contest name to a set of contest objectId strings. This way the error message can be of the format

Contest '${contestName}' appears in ${num} contests: $id1, $id2, ...., $idN

E.g.,

Contest 'Governor' appears in 2 contests: gov-NH, gov-VA